lib/instrument/transformer/nskeyed.js

```diff
@@ -367,13 +448,19 @@ class Archive {
       return this.encodeArray(obj, archiveObj);
     } else if (obj instanceof Set) {
       return this.encodeSet(obj, archiveObj);
+    } else if (obj instanceof XCTestConfiguration) {
+      return this.encodeXCTestConfiguration(obj, archiveObj);
     } else if (obj instanceof Object) {
       const objName = obj.constructor.name;
       // Only special class instance are useful, such as NSURL, NSUUID, NSDate.
       // And this class must also have the encodeArchive method
       if (objName in UNARCHIVE_CLASS_MAP) {
-        archiveObj.$class = this.uidForArchiver(objName);
         obj.encodeArchive(obj, new ArchivingObject(archiveObj, this));
+        if (obj.constructor.isNSObject) {
```

Comment:
same as above. Not sure about static field usage here. Also, why do we need to access the `constructor` property? I assume it should also accept `obj.isNSObject`